bids/analysis/variables.py

```diff
@@ -326,17 +326,17 @@ def load(self):
         run_trs = []
 
         for img_f in images:
-            f_ents = self.layout.files[img_f].entities
-            f_ents = {k: v for k, v in f_ents.items() if k in self.entities}
-            evf = self.layout.get(return_type='file', extensions='.tsv',
-                                  type='events', **f_ents)
+            evf = self.layout.get_events(img_f)
             if not evf:
                 raise ValueError("Could not find event file that matches %s." %
                                  img_f)
             tr = 1./self.layout.get_metadata(img_f)['RepetitionTime']
             run_trs.append(tr)
 
-            event_files.append((evf[0], img_f, f_ents))
+            f_ents = self.layout.files[evf].entities
```

Comment:
I'm not sure if this should be `files[evf]` or `files[img_f]`, by the way. `img_f` is going to be maximally specific, `evf` minimally.